library/psa_crypto_pake.c

```diff
@@ -0,0 +1,720 @@
+        PSA_ALG_IS_PAKE(cipher_suite->algorithm ) == 0 ||
+        ( cipher_suite->type != PSA_PAKE_PRIMITIVE_TYPE_ECC &&
+          cipher_suite->type != PSA_PAKE_PRIMITIVE_TYPE_DH ) ||
+        PSA_ALG_IS_HASH( cipher_suite->hash ) == 0 )
+    {
+        return( PSA_ERROR_INVALID_ARGUMENT );
+    }
+
+#if defined(MBEDTLS_PSA_BUILTIN_ALG_ECJPAKE)
+    if( cipher_suite->algorithm != PSA_ALG_JPAKE ||
+        cipher_suite->type != PSA_PAKE_PRIMITIVE_TYPE_ECC ||
+        cipher_suite->family != PSA_ECC_FAMILY_SECP_R1 ||
+        cipher_suite->bits != 256 ||
+        cipher_suite->hash != PSA_ALG_SHA_256 )
+    {
+        return( PSA_ERROR_NOT_SUPPORTED );
+    }
+
+    operation->alg = cipher_suite->algorithm;
+
+    mbedtls_ecjpake_init( &operation->ctx.ecjpake );
+
+    operation->state = PSA_PAKE_STATE_SETUP;
+    operation->sequence = PSA_PAKE_SEQ_INVALID;
+    operation->input_step = PSA_PAKE_STEP_X1_X2;
+    operation->output_step = PSA_PAKE_STEP_X1_X2;
+
+    operation->buffer = NULL;
+    operation->buffer_length = 0;
+    operation->buffer_offset = 0;
+
+    return( PSA_SUCCESS );
+#else
+    return( PSA_ERROR_NOT_SUPPORTED );
+#endif
+}
+
+psa_status_t psa_pake_set_password_key( psa_pake_operation_t *operation,
+                                        mbedtls_svc_key_id_t password )
+{
+    psa_status_t status = PSA_ERROR_CORRUPTION_DETECTED;
+    psa_key_attributes_t attributes = psa_key_attributes_init();
+    psa_key_type_t type;
+    psa_key_usage_t usage;
+
+    if( operation->alg == 0 ||
+        operation->state != PSA_PAKE_STATE_SETUP )
+    {
+        return( PSA_ERROR_BAD_STATE );
+    }
+
+    status = psa_get_key_attributes( password, &attributes );
+    if( status != PSA_SUCCESS )
+        return status;
+
+    type = psa_get_key_type( &attributes );
+    usage = psa_get_key_usage_flags( &attributes );
+
+    psa_reset_key_attributes( &attributes );
+
+    if( type != PSA_KEY_TYPE_PASSWORD &&
+        type != PSA_KEY_TYPE_PASSWORD_HASH )
+    {
+        return PSA_ERROR_INVALID_ARGUMENT;
+    }
+
+    if( usage == 0 ||
+        ( usage & PSA_KEY_USAGE_DERIVE ) == 0 )
+    {
+        return PSA_ERROR_NOT_PERMITTED;
+    }
+
+    operation->password = password;
+
+    return( PSA_SUCCESS );
+}
+
+psa_status_t psa_pake_set_user( psa_pake_operation_t *operation,
+                                const uint8_t *user_id,
+                                size_t user_id_len )
+{
+    if( operation->alg == 0 ||
+        operation->state != PSA_PAKE_STATE_SETUP )
+    {
+        return( PSA_ERROR_BAD_STATE );
+    }
+
+    if( user_id_len == 0 || user_id == NULL )
+        return PSA_ERROR_INVALID_ARGUMENT;
+
+    return( PSA_ERROR_NOT_SUPPORTED );
+}
+
+psa_status_t psa_pake_set_peer( psa_pake_operation_t *operation,
+                                const uint8_t *peer_id,
+                                size_t peer_id_len )
+{
+    if( operation->alg == 0 ||
+        operation->state != PSA_PAKE_STATE_SETUP )
+    {
```

Comment:
suggestion: braces not needed.

Reply:
same as https://github.com/Mbed-TLS/mbedtls/pull/5871/files/ac7885cd5f7c7fb6e3e934ead1850e1db240606a..530d163504143fd4a86b0dde02ca229a71f838c2#r892230939